keras/engine/training.py

```diff
@@ -2243,7 +2244,7 @@ def save_weights(self,
         return
     if save_format == 'h5':
       with h5py.File(filepath, 'w') as f:
-        hdf5_format.save_weights_to_hdf5_group(f, self.layers)
+        hdf5_format.save_model_weights_to_hdf5_group(f, self)
```

Comment:
Smart solution. Can you check it is backwards compatible, i.e. can the new logic load the old files?